.github/workflows/pr-test.yml

```
@@ -185,26 +186,27 @@ jobs:
       - name: Download Build Artifacts
         uses: actions/download-artifact@v4
         with:
-          name: tests-build-cache-${{ github.run_number }}
+          name: compile-cache-${{ github.run_number }}
           path: .nx
-      - name: Build
-        run: npm run compile
-      - name: Unit tests (Full)
-        if: matrix.code-coverage
-        run: npm run test
+      - name: Compile (Delta)
+        run: npm run compile:ci:affected
       - name: Unit tests (Delta)
-        if: ${{ !matrix.code-coverage }}
-        run: npm run test:ci:changed
-      - name: Report Coverage
-        if: ${{ matrix.code-coverage && !cancelled()}}
-        uses: codecov/codecov-action@v5
-        env:
-          CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
+        run: npm run test:ci:affected
+      - name: Test All Versions (Delta)
+        run: npm run test-all-versions:ci:affected
+      - name: Upload Test Artifacts
+        uses: actions/upload-artifact@v4
         with:
-          verbose: true
+          name: tests-coverage-cache-${{ github.run_number }}-${{ matrix.node }}
+          include-hidden-files: true
+          if-no-files-found: error
+          retention-days: 1
+          path: |
+            !node_modules
+            **/.nyc_output/**
```

There's [a bug in glob resolution of actions/toolkit](https://github.com/actions/toolkit/issues/657), which will walk the directory tree even if that path should already be excluded by the exclusion rule.

In the case of this PR, you can reduce the time to upload an artifact from ~15s to ~3s (not critical):
```suggestion
            packages/*/.nyc_output/**
            incubator/*/.nyc_output/**
```

In the modified tav version of this PR, where I used npm's isolated installs feature which produces deep directory structures, it took 10m to walk the directory tree! By changing the glob pattern to the one above, the step was executed in 3s instead.

---

@Ugzuzg thanks for the tip. I'll apply your 2nd suggestion :)

---

Looking at `package.json` I see incubator is not listed as part of the workspaces. I've did a quick search in the commit history but couldn't find `incubator` folder in workspaces. Apparently it wasn't even present when we use the old `lerna.json` conifg. You can check it at https://github.com/open-telemetry/opentelemetry-js-contrib/pull/1771

---

> So I'll keep incubator out for now and add it in a follow up PR. Will mention it in today's SIG

It might not be needed since that package is about to be moved from incubator to packages
https://github.com/open-telemetry/opentelemetry-js-contrib/pull/2993